File: solutions/cpp/all-your-base/1/all_your_base.cpp
#include "all_your_base.h"

namespace all_your_base {
    bool check_input(std::vector<unsigned int> input_digits, unsigned int input_base, unsigned int output_base) {
    if (input_base < 2 || output_base < 2)
        throw std::invalid_argument("Base must be larger than 1.");
    
    for (unsigned int digit : input_digits)
    {
        if (digit >= input_base)
            throw std::invalid_argument("Input digit cannot be equal to or larger than input base.");
    }

    if (input_digits.size() < 1)
        return true;
    return false;
}
    
    std::vector<unsigned int> convert(unsigned int input_base, std::vector<unsigned int> input_digits, unsigned int output_base) {
        std::vector<unsigned int> converted;
        if (check_input(input_digits, input_base, output_base))
            return converted;
        int number = 0;
        for (unsigned int i = 0; i < input_digits.size(); i++)
            number += input_digits[i] * std::pow(input_base, input_digits.size() - i - 1);
        
        if (number == 0)
            return converted;

        while (number > 0)
        {
            converted.insert(converted.begin(), number % (int)output_base);
            number /= (int)output_base;
        }
        
        return converted;
    }
}  // namespace all_your_base

File: solutions/python/all-your-base/1/all_your_base.py
def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")
    if output_base < 2:
        raise ValueError("output base must be >= 2")
    for digit in digits:
        if digit < 0 or digit >= input_base:
            raise ValueError("all digits must satisfy 0 <= d < input base")
    if digits == [0] or digits == []:
        return [0]
    number = 0
    for i in range(len(digits) - 1, -1, -1):
        current = 1
        for j in range(i):
            current *= input_base
        number += (digits[len(digits) - i - 1] * current)
    output_digits = []
    while number > 0:
        output_digits.append(number % output_base)
        number //= output_base
    output_digits.reverse()
    if (output_digits == []):
        return [0]
    return output_digits
